sentinel-core/src/test/java/com/alibaba/csp/sentinel/slots/block/degrade/DegradeTest.java

```diff
@@ -83,17 +84,38 @@ public void testExceptionRatioModeDegrade() throws Throwable {
         rule.setResource(key);
         rule.setTimeWindow(2);
         rule.setGrade(RuleConstant.DEGRADE_GRADE_EXCEPTION_RATIO);
+        rule.setMinRequestAmount(20);
 
-        when(cn.successQps()).thenReturn(8d);
 
-        // Will fail.
+        // Will true. While totalQps < minRequestAmount
+        when(cn.totalQps()).thenReturn(8d);
+        assertTrue(rule.passCheck(context, node, 1));
+
+        // Will true. While totalQps > minRequestAmount and exceptionCount < minRequestAmount
+        when(cn.totalQps()).thenReturn(21d);
+        when(cn.exceptionQps()).thenReturn(19d);
+        assertTrue(rule.passCheck(context, node, 1));
```

Comment:
There are some problems in the test case. Here `successQps` is not provided, so it's by default 0, which does not match the actual relation:

- in the same aligned statistic time window, success (aka. completed count) = exception count + non-exception count (realSuccess)

Reply:
> There are some problems in the test case. Here `successQps` is not provided, so it's by default 0, which does not match the actual relation:
> 
> * in the same aligned statistic time window, success (aka. completed count) = exception count + non-exception count (realSuccess)

Do you mean ?
```
            double realSuccess = success - exception;
            if (realSuccess <= 0 && exception < minRequestAmount) {
                return true;
            }
```
I am somewhat confused. My understanding is that success is counted when the request should respond successfully. The exception is counted when the program throws a unknown exception. Why do you need a realSuccess?  The number of success and excption should be independent of each other.

Reply:
The name `successQps` might be confusing here. The `successQps` in Sentinel actually means **complete** (i.e. passed from Sentinel checking, and has finished invocation, no matter it succeeded or failed logically), so the *success count* actually includes the *exception count*. The real success count (what you've understood) should be `success - exception`.

Reply:
understood, the latest code had commit.